### PR TITLE
Ready for building mvn artifactory for release to the public

### PR DESCRIPTION
**Pull request by shnooga**

### CODE REVIEW

File: api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import edu.wgu.osmt.db.PublishStatus.Archived
+import edu.wgu.osmt.db.PublishStatus.Unarchived
+import org.springframework.stereotype.Service
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.oauth2.core.user.OAuth2User
+import org.springframework.security.oauth2.jwt.Jwt
+
+
+@Service
+class AuthHelperService {
```
Comment (hikr17) [inline api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt:14]: Is this meant as a strategy pattern replacement for auth, or something else?

**Review by JohnKallies — APPROVED** [review]

File: api/src/main/kotlin/edu/wgu/osmt/RoutePaths.kt [hunk 2]
```diff
@@ -45,7 +45,7 @@ object RoutePaths {
         const val COLLECTION_ID = "collectionId"
     }
 
-    fun scrubForConfigure(routePath: String): String {
-        return routePath.replace("{uuid}", "*")
+    fun createRegex(routePath: String): String {
```
Comment (JohnKallies) [inline api/src/main/kotlin/edu/wgu/osmt/RoutePaths.kt:48]: This is a useful clarification, nice! Can we go all the way to "uuidRegex"?

File: api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt [hunk 3]
```diff
@@ -68,42 +65,47 @@ class SecurityConfig : WebSecurityConfigurerAdapter() {
...
-            .antMatchers(HttpMethod.GET, scrubForConfigure(TASK_DETAIL_TEXT)).permitAll()   // public csv results
+            .regexMatchers(HttpMethod.GET, createRegex(SKILL_DETAIL)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(COLLECTION_DETAIL)).permitAll()
+
+            .regexMatchers(HttpMethod.POST, createRegex(COLLECTION_SKILLS)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(COLLECTION_CSV)).permitAll()
+            .regexMatchers(HttpMethod.GET, createRegex(TASK_DETAIL_TEXT)).permitAll()   // public csv results
 
             // catch-all
-            .antMatchers("/**").permitAll()
```
Comment (JohnKallies) [inline api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt:106]: This line may still be needed in the open source version. They server the static client assets from the Spring app, and I think this is required. Try running a local Spring app with no ng serve and not permitAll(), see if you still get the front end?

File: api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillController.kt [hunk 4]
```diff
@@ -122,14 +122,18 @@ class RichSkillController @Autowired constructor(
     @ResponseBody
     fun updateSkill(
         @PathVariable uuid: String,
-        @RequestBody skillUpdate: ApiSkillUpdate,
-        @AuthenticationPrincipal user: Jwt?
+        @RequestBody skillUpdate: ApiSkillUpdate, @AuthenticationPrincipal user: Jwt?
     ): ApiSkill {
+
+        if (authHelperService.hasRole(appConfig.roleCurator) && !authHelperService.isArchiveRelated(skillUpdate.publishStatus)) {
```
[the remark this answers is not in the document]
Comment (shnooga) [reply api/src/main/kotlin/edu/wgu/osmt/richskill/RichSkillController.kt:128]: Need clarifications with JK

File: api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import edu.wgu.osmt.db.PublishStatus
+import edu.wgu.osmt.db.PublishStatus.Archived
+import edu.wgu.osmt.db.PublishStatus.Unarchived
+import org.springframework.stereotype.Service
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.oauth2.core.user.OAuth2User
+import org.springframework.security.oauth2.jwt.Jwt
+
+
+@Service
```
[the remark this answers is not in the document]
Comment (shnooga) [reply api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt:13]: Done deal.

File: api/src/main/kotlin/edu/wgu/osmt/RoutePaths.kt (hunk 2 above)
Comment (shnooga) [reply api/src/main/kotlin/edu/wgu/osmt/RoutePaths.kt:48]: Changing to createUuidRegex()

File: api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt (hunk 3 above)
Comment (shnooga) [reply api/src/main/kotlin/edu/wgu/osmt/security/SecurityConfig.kt:106]: We are able to run the UI on our local with no "ng serve". Need clarification/confirmation with JK

File: api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt (hunk 1 above)
Comment (shnooga) [reply api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt:14]: It's just a helper class for OAuth related stuff.

Comment (hikr17) [reply api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt]: Ya, I get that.  My question was more about the intention and form.  How is the code meant to be replaced or overridden from opensource by WGU code?

Comment (JohnKallies) [reply api/src/main/kotlin/edu/wgu/osmt/security/AuthHelperService.kt]: @hikr17 This class won't be overridden in WGU. It's just a class to stash a few methods. they are under unit test.

The SecurityConfig will be overridden, but this is different.

**Pull request merged**